File: benchmarks/oopsla25/subjects/044.c
#include <assert.h>
#include <fcntl.h>
#include <stdio.h>
#include <stdlib.h>
#include <string.h>
#include <unistd.h>

void f044(const char *s) {
    size_t n = strlen(s) - 2;
    char c1 = s[n];
    char c2 = s[n + 1];
    assert(n == 0);
}

int main(int argc, char *argv[]) {
  char my_string[10240] = {0};
  if (argc == 1) {
    read(fileno(stdin), my_string, 10240);
  } else {
    int fd = open(argv[1], O_RDONLY);
    read(fd, my_string, 10240);
    close(fd);
  }
  f044(my_string);
  return 0;
}

File: benchmarks/oopsla25/subjects/044.py
#!/usr/bin/env python3

import sys

def f044(s: str):
  n = len(s) - 2
  c1 = s[n]
  c2 = s[n+1]
  assert n == 0

if __name__ == '__main__':
  my_string = ""
  if len(sys.argv) == 1:
    my_string = sys.stdin.read()
  else:
    fd = open(sys.argv[1], 'r')
    my_string = fd.read()
    fd.close()
  
  f044(my_string)
